src/actions/guides/database/callbacks_and_validations.cr

```diff
@@ -0,0 +1,320 @@
+    | validate_size_of            | ensure a string is within a length range               | `validate_size_of username, min: 3, max: 11`                     |
+    | validate_uniqueness_of      | ensure a value doesn't exist in the database already   | `validate_uniqueness_of email, query: UserQuery.new.email`       |
+    | validate_at_most_one_filled | check that only 1 of the attributes has a value        | `validate_at_most_one_filled choice1, choice2, choice3`          |
+    | validate_exactly_one_filled | ensure exactly 1 attribute has a value                 | `validate_exactly_one_filled photo_id, video_id`                 |
+    | validate_numeric            | ensure a number is within the range specified          | `validate_numeric age, greater_than: 20, less_than: 30`          |
+
+    > Note: non-nilable (required) fields automatically use `validate_required`.
+    > They will run after all other `before_save` callbacks have run. This way data
+    > with missing fields will never be sent to the database.
+
+    ### Using validations
+
+    All of the validations require an `Avram::Attribute` to be passed in.
+
+    You can use validations inside of `before_save` callbacks:
+
+    ```crystal
+    class SaveUser < User::SaveOperation
+      permit_columns name, terms_of_service, age
+      attribute password : String
+      attribute password_confirmation : String
+
+      before_save do
+        validate_required name
+        validate_confirmation_of password, with: password_confirmation
+        validate_acceptance_of terms_of_service
+        validate_inclusion_of age, in: [30, 40, 50]
+        validate_uniqueness_of name
+        # Alternatively, pass optional second argument to use a custom query
+        validate_uniqueness_of name, query: UserQuery.new.name.lower
+        # Showing these version as an example
+        # You would not want all three of these on a real form
+        validate_size_of name, is: 4 # Name must be 4 characters long
+        validate_size_of name, min: 4 # Can't be too short
+        validate_size_of name, max: 8 # Must have a short name
+      end
+
+      before_save reject_scary_monsters
+
+      def reject_scary_monsters
+        if name.value == "Skeletor"
+          name.add_error "Mmmyyaahhh!"
+        end
+      end
+    end
+    ```
+
+    ### Custom error messages
+
+    When a validation fails, an error message will be applied to the attribute automatically. For example,
+    the `validate_required` will return `"is required"`, and the `validate_acceptance_of` will return `"must be accepted"`.
+
+    You can pass a `message` option to any of the attributes to define your own custom error message.
+
+    ```crystal
+    before_save do
+      validate_required name, message: "se requiere"
+      validate_confirmation_of email, with: email_confirmation, message: "debe coincidir con"
+    end
+    ```
+
+    ## What are attributes?
```

Comment:
This section is technically duplicated between the two guides; however, I think it's fine since you need to know about attributes when doing validations, but you also need to know about them when talking about `attribute`, `permit_columns`, using `changed?`,etc... We could break them out in to their own guide later.